readme.md

```diff
@@ -1,29 +1,48 @@
 <!-- default badges list -->
-![](https://img.shields.io/endpoint?url=https://codecentral.devexpress.com/api/v1/VersionRange/255628686/22.2.5%2B)
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T967905)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
-*Files to look at*:
 
-* [LiveSummary.cs](./CS/EFCore/NonPersReloadEF/NonPersReloadEF.Module/BusinessObjects/LiveSummary.cs)
-* [NonPersistentObjectBase.cs](./CS/EFCore/NonPersReloadEF/NonPersReloadEF.Module/BusinessObjects/NonPersistentObjectBase.cs)
-* [Module.cs](./CS/EFCore/NonPersReloadEF/NonPersReloadEF.Module/Module.cs)
 
+# XAF - How to refresh non-persistent objects and reload nested persistent objects
 
-# How to refresh Non-Persistent Objects and reload nested Persistent Objects
 
-## Scenario
+It is often necessary to undo changes made to [non\-persistent objects](https://docs.devexpress.com/eXpressAppFramework/116516/concepts/business-model-design/non-persistent-objects?v=20.1) in a view. If a non-persistent object has links to persistent objects, it is often required to reload these linked objects as well. However, by default, the built-in **Refresh** action has no effect in these scenarios. This example shows how to refresh data in such cases.
 
-It is often required to cancel changes made to [Non\-Persistent Objects](https://docs.devexpress.com/eXpressAppFramework/116516/concepts/business-model-design/non-persistent-objects?v=20.1) in a view. When a Non-Persistent object has links to Persistent Objects, it is often required to reload these linked objects too. However, the built-in Refresh action has no effect in these scenarios by default.
+<kbd>![devenv_8Nt2WqQLBv](https://github.com/DevExpress-Examples/XAF_Non-Persistent-Objects-Reloading-Demo/assets/14300209/0ad2e8ab-ac17-4844-a9d1-a369afe6beca)</kbd>
 
-## Solution
+> [!WARNING]
+> We created this example for demonstration purposes and it is not intended to address all possible usage scenarios with it.
+> If this example does not have certain functionality or you want to change its behavior, you can extend this example as needed. Please note that this can be a complex task that requires good knowledge of XAF: [UI Customization Categories by Skill Level](https://www.devexpress.com/products/net/application_framework/xaf-considerations-for-newcomers.xml#ui-customization-categories). You will likely need to research how our components work under the hood. Refer to the following help topic for more information: [Debug DevExpress .NET Source Code with PDB Symbols](https://docs.devexpress.com/GeneralInformation/403656/support-debug-troubleshooting/debug-controls-with-debug-symbols).
```

Comment:
```suggestion
> If this example does not have certain functionality or you want to change its behavior, you can extend this example. This can be a complex task that requires good knowledge of XAF: [UI Customization Categories by Skill Level](https://www.devexpress.com/products/net/application_framework/xaf-considerations-for-newcomers.xml#ui-customization-categories), and you may need to research how our components work. Refer to the following help topic for more information: [Debug DevExpress .NET Source Code with PDB Symbols](https://docs.devexpress.com/GeneralInformation/403656/support-debug-troubleshooting/debug-controls-with-debug-symbols).
```